.github/workflows/python-build.yml

```diff
@@ -1,6 +1,14 @@
 name: Python Tests
 run-name: Run Tests by ${{ github.actor }}
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+  pull_request:
+  release:
+    types: [published]
+  workflow_dispatch:
+permissions:
+  id-token: write
```

Comment:
Giving OIDC to random software in all the jobs is dangerous. I filed an issue about the problem in the publishing job. This makes the exposure even worse.

Reply:
This is not necessary to have ` release published` trigger as well, since currently the tag is the main trigger for uploading a package to pypi. Creating the github release has to be done manually though.